### DISCUSSION TITLE
chart title from the datavalues

### DISCUSSION THREAD

**Discussion by Joerg55** [vega/vega-lite#8788] [Q&A]
Hi community,

i want to generate the title of my chart from the data values. For example the title of this chart should be 'charttitle':

`{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
 "data": {
    "values": [
      {"quater":"1", "sales":"100", "title":"charttitle"},
      {"quater":"2", "sales":"200", "title":"charttitle"},
      {"quater":"3", "sales":"300", "title":"charttitle"},
      {"quater":"4", "sales":"400", "title":"charttitle"}
    ]
  },
  
  "mark": "bar",
  "encoding": {
    "x": {"field": "quater", "type":"ordinal"},
    "y": {"field": "sales", "type": "quantitative"},
  },
  "title": ???
}`

How can I do that?

**Comment by lutzbickhardt** [comment]
I cannot fiend any indication in the $schema that this is allowed. Further it would be nice if one could:
 "encoding": {
    "x": {"field": "time", "type": "temporal", 
        "axis": {"format" : "%H:%M", "title": {"field": "yearmonthdate"}}
    }  
  }

or even better

 "encoding": {
    "x": {"field": "time", "type": "temporal", 
        "axis": {"format" : "%H:%M", "title": {["field": "year", "-", "field": "month"}, "-"+ "field": "day"}]}
    }  
  }

so one could indicate YYYYMMDD of the hours and minutes on th x-axis.

**Reply by lutzbickhardt** [reply]
sorry, I mean:
"axis": {"format" : "%H:%M", "title": {["field": "year", "-", "field": "month"}, "-", "field": "day"}]}
